lib/zclient.h

```diff
@@ -348,6 +350,19 @@ struct zclient {
 	void (*zebra_connected)(struct zclient *);
 	void (*zebra_capabilities)(struct zclient_capabilities *cap);
 
+	/*
+	 * match -> is the prefix that the calling daemon asked to be matched
+	 * against.
+	 * nhr->prefix -> is the actual prefix that was matched against in the
+	 * rib itself.
+	 *
+	 * This distinction is made because a LPM can be made if there is a
+	 * covering route.  This way the upper level protocol can make a
+	 * decision point about whether or not it wants to use the match or not.
+	 */
+	void (*nexthop_update)(struct vrf *vrf, struct prefix *match,
```

Comment:
'nexthop_update' seems a bit ambiguous - now that we have nexthop-groups etc. could it be 'nht_update' or 'rnh_update' ?

Reply:
Happy to rename it, I just "inherited" the name from `zapi_nexthop_update_decode`. Donald and I also have renaming the entirety of NHT on the TODO list (because the name is kinda misleading), maybe it doesn't matter at this point?

Reply:
(I don't think `rnh` is a good choice of name, this isn't about recursive resolution; `nht` is off in general since this isn't a mechanism for specifically tracking nexthops, it tracks prefixes in general… cf. discussion in #development)